src/backend/distributed/executor/executor.c

```diff
@@ -1230,6 +1233,28 @@ ManageWorkerPool(WorkerPool *workerPool)
 	int newConnectionCount = 0;
 	int connectionIndex = 0;
 
+	/* we should always have more (or equal) active connections than idle connections */
+	Assert (activeConnectionCount >= idleConnectionCount);
+
+	/* we should never have less than 0 connections ever */
+	Assert (activeConnectionCount >= 0 && idleConnectionCount >= 0);
+
+	TimestampTz poolCreationTime = workerPool->poolCreationTime;
+	TimestampTz now = GetCurrentTimestamp();
+
+	if (activeConnectionCount == 0 &&
```

Comment:
Hmm, we should also consider the case when `force_max_query_parallelization = true`. In that case, we should probably check `len(workerPool->sessionList) != activeConnectionCount ` instead of `activeConnectionCount == 0`.